### DISCUSSION TITLE
Switching to a newer standard?

### DISCUSSION THREAD

**Discussion by realstealthninja** [TheAlgorithms/C-Plus-Plus#2744] [Q&A]
# Its been 13 years since C++11

Its fair to say that a newer standard will help as C++ has grown older, New features are added.
which make the development experience not only easier but more importantly enjoyable.

Please weigh in on if we should switch and what standard to switch to.

**Comment by SivaPA08** [comment]
C++11 is 13 years old, so it might be time to upgrade. Newer standards offer useful features that make coding easier and more enjoyable. For example, C++17 lets you work with data more simply and can speed up your code with parallel algorithms. C++20 goes even further by adding modules for better code organization and concepts to simplify templates. Ultimately, the best choice depends on you and your project's needs. If your tools support it, switching to C++20 is a great upgrade; otherwise, C++17 is a solid and safe option.

**Comment by Nithwin** [comment]
I strongly recommend upgrading to **C++20**.

While C++17 offers great improvements (`std::optional`, `std::string_view`), jumping straight to **C++20** is the best move for an Algorithms repository for one specific reason: **Concepts**.

Since this repo is heavily templated, C++20 Concepts would allow us to replace complex `SFINAE` or `std::enable_if` syntax with readable constraints. This is crucial for an educational project because it makes compiler errors much easier for students to understand.

**Example Upgrade:**
Instead of a verbose template like this (C++11/14):
```cpp
template <typename T>
typename std::enable_if<std::is_integral<T>::value, bool>::type
is_even(T number) { ... }
```
We can simply write (C++20):
```cpp
template <std::integral T>
bool is_even(T number) { ... }
```
